src/NUnitFramework/framework/Compatibility/System.Collections.Concurrent/ConcurrentQueue.cs

```diff
@@ -36,31 +35,64 @@ namespace System.Collections.Concurrent
     /// </summary>
     /// <typeparam name="T">Specifies the type of elements in the queue.</typeparam>
     /// <remarks>
-    /// All public and protected members of <see cref="ConcurrentQueue{T}"/> are thread-safe and may be used
+    /// All public  and protected members of <see cref="ConcurrentQueue{T}"/> are thread-safe and may be used
     /// concurrently from multiple threads.
     /// </remarks>
-	[System.Diagnostics.DebuggerDisplay ("Count={Count}")]
-	[System.Diagnostics.DebuggerTypeProxy (typeof (CollectionDebuggerView<>))]
-	internal class ConcurrentQueue<T> : IProducerConsumerCollection<T>, IEnumerable<T>, ICollection,
-	                                  IEnumerable
-	{
-		class Node
-		{
-			public T Value;
-			public Node Next;
-		}
-
-		Node head = new Node ();
-		Node tail;
-		int count;
+    [ComVisible(false)]
+    [DebuggerDisplay("Count = {Count}")]
+    [DebuggerTypeProxy(typeof(SystemCollectionsConcurrent_ProducerConsumerCollectionDebugView<>))]
+    [HostProtection(Synchronization = true, ExternalThreading = true)]
+    [Serializable]
+    public class ConcurrentQueue<T> : IProducerConsumerCollection<T>//, IReadOnlyCollection<T>
```

Comment:
Second change from Mono, we don't use the `IReadOnlyCollection<T>` interface, so I didn't bother to bring it across.